### PR TITLE
Update: add ability to parse optional properties in typedefs (refs #5)

### PR DESCRIPTION
**Pull request by ikokostya**
This is rebased to master pull request https://github.com/eslint/doctrine/pull/162. Unrelated changes in `CHANGELOG.md` and `package.json` are removed.

### CODE REVIEW

File: lib/doctrine.js [hunk 1]
```diff
@@ -255,9 +259,11 @@
                 return utility.throwError('Braces are not balanced');
             }
 
-            if (isParamTitle(title)) {
+            if (isParamTitle(title) || (sloppy && isAllowedOptional(title))) {
```
Comment (nzakas) [inline lib/doctrine.js]: Can't this just be replaced with `if (isAllowedOptional(title))`?

**Comment by nzakas** [comment]
Overall looks good, just one question (left inline).

File: lib/doctrine.js (hunk 1 above)
Comment (ikokostya) [inline lib/doctrine.js]: I think yes. All tests passed.

**Comment by nzakas** [comment]
Thanks! I'll do a new release with this.

**Pull request merged**